config.go

```diff
@@ -30,8 +37,19 @@ func parsePluginConfiguration(data []byte) (pluginConfiguration, error) {
 	jsonData := gjson.ParseBytes(data)
 	config.rules = []string{}
 	jsonData.Get("rules").ForEach(func(_, value gjson.Result) bool {
-		config.rules = append(config.rules, value.String())
+		config.rules = append(config.rules, handlePluginConfigurationKeywords(value.String()))
 		return true
 	})
 	return config, nil
 }
+
+// handlePluginConfigurationKeywords replaces high level configuration keywords
+// with the internal paths
+func handlePluginConfigurationKeywords(configLine string) string {
+	for k, v := range keywordsDict {
+		re := regexp.MustCompile(`(?i)include ` + k)
```

Comment:
In here we are compiling all rules every time per line. Since `keywordsDict` is fixed we probably compile them in the var:

```
var keywordsDict = map[string]*regexp.Regexp{
	"@recommended-conf": regexp.MustCompile("(?i)include coraza.conf-recommended.conf"),
	...
}
```

Reply:
Thanks JC, the original idea was to keep the dictionary as clear as possible, but in that way, rules are really getting compiled too many times

Reply:
Turning the dictionary into the following
```
var keywordsDict = map[*regexp.Regexp]string{
	regexp.MustCompile("(?i)include @recommended-conf"): "coraza.conf-recommended.conf",
	regexp.MustCompile("(?i)include @crs-conf"):         "crs-setup.conf.example",
	regexp.MustCompile("(?i)include @owasp_crs"):        "crs",
}
```
leads to an error at compile time:
```
panic: cannot convert pointer value to byte

goroutine 269 [running]:
github.com/tinygo-org/tinygo/interp.rawValue.bytes(...)
	/tinygo/interp/memory.go:727
github.com/tinygo-org/tinygo/interp.rawValue.Uint(...)
	/tinygo/interp/memory.go:735
github.com/tinygo-org/tinygo/interp.(*runner).run(0x4004409f40, 0x400dbe9130, {0x400c285010, 0x4, 0x4015525770?}, 0x4016255140, {0x4014325db0, 0xc})
	/tinygo/interp/interpreter.go:866 +0x18d0
github.com/tinygo-org/tinygo/interp.(*runner).run(0x4004409f40, 0x400d574320, {0x400b3ac010, 0x4, 0x400b8eea08?}, 0x4007f698f0, {0x4014325d70, 0x8})
	/tinygo/interp/interpreter.go:547 +0x6404
github.com/tinygo-org/tinygo/interp.(*runner).run(0x4004409f40, 0x400b6f97c0, {0x0, 0x0, 0x4007d3d3e0?}, 0x0, {0x7c32a8, 0x4})
	/tinygo/interp/interpreter.go:547 +0x6404
github.com/tinygo-org/tinygo/interp.Run({0xffff40559040?}, 0x0?, 0x0?)
	/tinygo/interp/interp.go:117 +0x500
github.com/tinygo-org/tinygo/builder.optimizeProgram({0xffff782dcaf8?}, 0x40001fc780)
	/tinygo/builder/build.go:1049 +0x34
github.com/tinygo-org/tinygo/builder.Build.func5(0x4007e17740)
	/tinygo/builder/build.go:555 +0x49c
github.com/tinygo-org/tinygo/builder.runJob(0x4007e177a0, 0xd?)
	/tinygo/builder/jobs.go:222 +0x48
created by github.com/tinygo-org/tinygo/builder.runJobs
	/tinygo/builder/jobs.go:123 +0x448
Error: running "docker run --pull=always --rm -v /Users/user/Repo/coraza-proxy-wasm:/src ghcr.io/corazawaf/coraza-proxy-wasm/buildtools-tinygo:main bash -c
cd /src && \
tinygo build -gc=none -opt=2 -o build/mainraw.wasm -scheduler=none -target=wasi -tags='custommalloc'" failed with exit code 2
```
It is something that I'm not experiencing inside the [Go Playground](https://go.dev/play/p/LCP-lxvIqK9). Am I missing something obvious? 

As a workaround, I committed a different implementation. The dictionary ends up being quite clear and regexes are compiled just once in `parsePluginConfiguration`.
```
var keywordsDict = []keyword{
	{key: "@recommended-conf", tran: "coraza.conf-recommended.conf"},
	{key: "@crs-conf", tran: "crs-setup.conf.example"},
	{key: "@owasp_crs", tran: "crs"},
}
```

Could it be okay?

Reply:
I was wondering why the map uses the regex as key instead of the filename.

Reply:
Logically the keywords that we are looking for inside the config are `@recommended-conf`, `@crs-conf`, and `@owasp_crs`. I thought it was more intuitive to consider them as the keys, while their translation as path or filename is the value